src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py

```diff
@@ -783,3 +822,169 @@ def __call__(
             return (image, has_nsfw_concept)
 
         return StableDiffusionPipelineOutput(images=image, nsfw_content_detected=has_nsfw_concept)
+
+    # yiyi notes and todo: put here for testing for now, make it mixin later 
```

Comment:
@sayakpaul 
I think a major to-do left is to refactor the `_set_ip_adapter` and `load_ip_adapter` and make a mixin out of it :)

Reply:
Done.